packages/shared/src/hooks/feed/useAdvancedSettings.ts

```diff
@@ -29,34 +30,36 @@ export const useAdvancedSettings = (
     [feedSettings?.advancedSettings],
   );
 
-  const onToggleSettings = useCallback(
-    (id: number, defaultEnabledState: boolean) => {
-      if (alerts?.filter && user) {
-        updateAlerts({ filter: false });
-      }
-
-      const enabled = !(selectedSettings[id] ?? defaultEnabledState);
-
+  const onUpdateSettings = useCallback(
+    (id: number, enabled: boolean) => {
       trackEvent({
         event_name: `toggle ${enabled ? 'on' : 'off'}`,
         target_type: 'advanced setting',
         target_id: id.toString(),
         extra: JSON.stringify({ origin: 'advanced settings filter' }),
       });
 
-      updateAdvancedSettings({
+      return updateAdvancedSettings({
         advancedSettings: [{ id, enabled }],
       });
     },
-    [
-      alerts?.filter,
-      selectedSettings,
-      trackEvent,
-      updateAdvancedSettings,
-      updateAlerts,
-      user,
-    ],
+    [trackEvent, updateAdvancedSettings],
+  );
+
+  // We still need to support this function as this is used for experiments
```

Comment:
We can't migrate both into one?

Reply:
We can, probably. Just not sure if we should do it as we've had two experiments on separate occasions on this part. Now that we have new advanced settings, we might again introduce the same experiment. Wdyt?

Reply:
Hmm yeah let's not complicate it for now :) 
We can adjust later as well.